src/main/kotlin/org/rust/cargo/runconfig/CargoRunState.kt

```diff
@@ -42,6 +42,8 @@ class CargoRunState(
             // Even though default system encoding is usually not UTF-8 on windows,
             // most Rust programs are UTF-8 only.
             .withCharset(Charsets.UTF_8)
+            .withParameters("--") // Tell cargo remaining args are for the debug target
+            .withParameters(commandLine.getRunArguments())
```

Comment:
I think these two lines might not be needed here? All arguments should be added as part of `additionalArguments` inside `toColoredCommandLine`.

Reply:
- Removed those lines
- Wrote a unit test
- Verified in CLion (debug and run)
- Rebased
- Pushed